File: Visitor/PHP/CreditCard.php
<?php

require_once "IPayment.php";
require_once "CPU.php";
require_once "VGA.php";

class CreditCard implements IPayment {
    public function visit($component) {
        if ($component instanceof CPU) {
            echo "Purchase CPU with Credit Card\n";
        } elseif ($component instanceof VGA) {
            echo "Purchase VGA with Credit Card\n";
        }
    }
}

File: Visitor/Golang/CreditCard.go
package main

import "fmt"

type CreditCard struct {}

func (visitor *CreditCard) visitCPU(component *CPU)  {
	fmt.Println("Purchase CPU with Credit Card")
}

func (visitor *CreditCard) visitVGA(component *VGA)  {
	fmt.Println("Purchase VGA with Credit Card")
}